#2 and #3 Reader Page Turn animation and book shadowing

Tim-M-Matthews opened the pull request:
Adds animation to the R2D2BC reader via changes to IFrameNavigator.ts and a new _rif.scss file

This closes #2 
This closes #3 

## To Test

- [x] Checkout this branch
- [x] Download the epub files from [this dropbox link](https://www.dropbox.com/scl/fo/775e28nwjilc1g9yov3g0/AM2Qdax9vhsbrhiYD2Yg3B4?rlkey=tz7x406b72r2ncn2vaotqqhvz&e=2&st=4xpk0lgk&dl=0) and add them to the epubs folder
- [x] Test paging forward and backward and confirm it works for you consistently across books. Be sure to try both Lola's Lumpia and one from the list as they are different aspect ratios.
- [x] Pay close attention to the center shadow to make sure z-positioning is not an issue between it and the moving pages

Tim-M-Matthews committed:
#2 and #3 add basic stylistic overlay and page turning animations

Tim-M-Matthews committed:
#2 and #3 refine UX across other ebooks

maddybvh reviewed (changes requested):
Hi @Tim-M-Matthews ! 

- [x] There are more code changes in `IFrameNavigator.ts` than feels maintainable because we're trying to keep the backwards compatibility with the source code. Can you review all code changes, remove anything that isn't strictly necessary, and add a note on why something is necessary? Please avoid committing whitespace changes for this reason, too. I know the formatting is annoying.
- [x] We'll need to maintain the loading state for non-fixed books, so please uncomment that code and be more surgical. Here's a [sample non-fixed epub](https://drive.google.com/file/d/1KlBo8QYygm8-n8ptD14m8CuxkHcHs26P/view?usp=drive_link).

Tim-M-Matthews committed:
#2 and #3 document savas code changes and localize to two-column layout

Tim-M-Matthews committed:
#2 and #3 remove add'l whitespace

Tim-M-Matthews commented:
@maddybvh thanks for the guidance here! Here's what I changed

1. I added `Savas` and a message in comments above all changes to IFrameNavigator.ts explaining what they do
2. I removed all whitespace changes
3. I refactored back to the original code in a few cases where it wasn't strictly necessary
4. I added some handling in the js and css to make sure changes wouldn't happen on epubs such as the one you included

Tim-M-Matthews committed:
#2 and #3 removing whitespace cont'd

Tim-M-Matthews committed:
#2 and #3 restore original iframe clip path

Tim-M-Matthews committed:
#2 and #3 restore original opacity settings for showIframeContents

maddybvh reviewed (changes requested):
I'm not sure if your changes caused regressions, but I'm noticing some new issues:

- [x] Please clean up clones after you're done using them:
![Screenshot 2025-03-26 at 12 36 21 PM](https://github.com/user-attachments/assets/f7a35d1a-8a7c-4529-978b-1ee2fb9ccb0d)
- [x] This might be solved by the request above, but I'm seeing some weird behavior where the page gets turned to the left of the book bounds
https://github.com/user-attachments/assets/624ad12a-c047-4b93-b32e-87563a312120
- [x] It seems like there's only boxshadow on the right-side page.
<img width="1428" alt="Screenshot 2025-03-26 at 12 39 20 PM" src="https://github.com/user-attachments/assets/6de347e7-131c-46d5-a346-9c0d89f1d024" />

Tim-M-Matthews committed:
#2 and #3 fix syntax of resetIframes call

Tim-M-Matthews commented:
@maddybvh ah yes, that was all caused by one final tweak to minimize code changes that led to a syntax error. Working for me now

maddybvh reviewed (approved):
Thanks @Tim-M-Matthews ! That white flash on page turn is pretty annoying. I created https://github.com/savaslabs/R2D2BC/issues/7 to follow up on it. If you end up with free time, please do keep pushing on that.
